lasagne/layers/conv.py

```diff
@@ -199,12 +198,14 @@ def __init__(self, incoming, num_filters, filter_size, stride=1,
         self.num_filters = num_filters
         self.filter_size = as_tuple(filter_size, 1)
         self.stride = as_tuple(stride, 1)
-        self.border_mode = border_mode
+        self.pad = pad
         self.untie_biases = untie_biases
         self.convolution = convolution
 
-        if self.border_mode not in ['valid', 'full', 'same']:
-            raise RuntimeError("Invalid border mode: '%s'" % self.border_mode)
+        if self.pad not in ['valid', 'full', 'same', 0]:
+            raise RuntimeError("Invalid pad: '%s'" % self.pad)
+        if self.pad == 0:
+            self.pad = 'valid'
```

Comment:
I'd probably do it the other way around, since officially (following the docstring), we only support `{0, 'full', 'same'}`. That will complicate the `self.convolution` call a little (`border_mode=pad or 'valid'`), but it might be surprising to have `Conv1DLayer.pad` be set to `valid` when you set it to `0`.